storage/cloud-client/hmac_test.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for hmac.py. Requires GOOGLE_CLOUD_PROJECT (valid project) and
+HMAC_KEY_TEST_SERVICE_ACCOUNT (valid service account email) env variables to be
+set in order to run.
+"""
+
+
+import os
+
+from google.cloud import storage
+import pytest
+
+import hmac
+
+
+PROJECT_ID = os.environ['GOOGLE_CLOUD_PROJECT']
+SERVICE_ACCOUNT_EMAIL = os.environ['HMAC_KEY_TEST_SERVICE_ACCOUNT']
+
+
+@pytest.fixture()
+def new_hmac_key():
+    """
+    Fixture to create a new HMAC key, and to guarantee all keys are deleted at
+    the end of each test.
+    """
+    hmac_key = hmac.create_key(PROJECT_ID, SERVICE_ACCOUNT_EMAIL)
+    yield hmac_key
+    storage_client = storage.Client(project=PROJECT_ID)
```

Comment:
We probably want to make this it's own fixture and scope it to `module` to avoiding initializing a new one every time.

Reply:
Are you referring specifically to storage_client? I could do this, but I guess I wasn't thinking of it as a resource that is expensive to initialize.

Reply:
Other tests such as https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/storage/cloud-client/acl_test.py and https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/storage/cloud-client/bucket_policy_only_test.py seem to use a similar approach with including storage.client in one function-level fixture.

Reply:
Yeah - some of our clients launch threads and tie up other resources (which is why I asked about `close()` elsewhere). This client doesn't look too heavy, but there are a [couple of network calls for oath and an http session being created](https://github.com/googleapis/google-cloud-python/blob/db481bfdd6816d020d99df0d4caa307358ab1141/core/google/cloud/client.py#L126), so it's probably worthwhile to reuse. 

I'm actually not sure if you can nest fixtures, but if it doesn't work and since you don't have to close it, you can probably just move it to a module level variable.

Reply:
Cool, I'll try this. Since many tests in this repo use the storage client I would move this to a file called shared_fixtures or something.

Reply:
You can nest fixtures, but I decided to go for the module-level variable because it seems pretty safe and it's less confusing to read.